cmd/otel-allocator/collector/collector.go

```diff
@@ -43,110 +43,102 @@ var (
 	})
 )
 
-type Client struct {
-	log       logr.Logger
-	k8sClient kubernetes.Interface
-	close     chan struct{}
+type CollectorWatcher struct {
```

Comment:
can we name this Watcher? No need to include the package name i.e. `collector.Watcher` vs `collector.CollectorWatcher`

Reply:
Sure.